apps/aehttpclient/lib/client.ex

```diff
@@ -44,14 +44,17 @@ defmodule Aehttpclient.Client do
 
   def get(uri, identifier) do
     case(HTTPoison.get(uri)) do
-      {:ok, %{body: body, status_code: 200}} ->
+      {:ok, %{body: body, headers: headers, status_code: 200}} ->
         case(identifier) do
           :block ->
             response = Poison.decode!(body, as: %Block{}, keys: :atoms!)
             {:ok, response}
           :info ->
             response = Poison.decode!(body, keys: :atoms!)
-            {:ok, response}
+            {_, server} = Enum.find(headers, fn(header) ->
+              header == {"server", "aehttpserver"} end)
```

Comment:
We force `server` header to be equal "aehttpserver" also here. I wonder: what happens if it's not? Do we get an exception that then goes back to `Aecore.Peers.Worker` and causes it to restart?

Reply:
@d-velev what do you think?